app/scripts/metamask-controller.js

```diff
@@ -943,15 +913,33 @@ export default class MetamaskController extends EventEmitter {
       removeIdentity: this.preferencesController.removeAddress.bind(
         this.preferencesController,
       ),
-      setAccountLabel: this.preferencesController.setAccountLabel.bind(
-        this.preferencesController,
-      ),
-      setSelectedAddress: this.preferencesController.setSelectedAddress.bind(
-        this.preferencesController,
-      ),
-      syncIdentities: this.preferencesController.syncAddresses.bind(
-        this.preferencesController,
-      ),
+      setAccountLabel: (address, label) => {
+        const accountToBeNamed = this.accountsController
```

Comment:
I feel like this functionality should be inside the accountsController, with a `find()` method

Reply:
Made a change on [509c604](https://github.com/MetaMask/metamask-extension/pull/21437/commits/509c60451fd15d3b53eed306a62d24ab81c52f78) to use `getAccountByAddress`